scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DatasetService.java

```diff
@@ -0,0 +1,193 @@
+                          DatasetRepository datasetRepository,
+                          DataFileRepository dataFileRepository) {
+        this.fileSystemService = fileSystemService;
+        this.datasetRepository = datasetRepository;
+        this.dataFileRepository = dataFileRepository;
+    }
+
+    /**
+     * Creates the dataset in the filesystem, then in the database.
+     *
+     * @param name The name of the Dataset.
+     * @param user The IUser who owns the dataset
+     * @throws DatasetStorageException Exception thrown if dataset cannot be created.
+     */
+    @Override
+    public void createDataset(String name, IUser user) throws DatasetStorageException {
+
+        Dataset dataset = new Dataset();
+        dataset.setName(name);
+        dataset.setUser(user);
+        Dataset savedDataset = datasetRepository.save(dataset);
+        Long savedDatasetId = savedDataset.getId();
+
+        // create directories on the file system (using Dataset id)
+        try {
+            fileSystemService.addDatasetDirectories(String.valueOf(savedDatasetId),
+                    user.getUsername());
+        } catch (DatasetStorageException e) {
+            throw new DatasetStorageException(
+                    "An IOException prevented the Dataset from being created", e);
+        }
+    }
+
+    /**
+     * Edits the dataset name in the database (Dataset path uses id).
+     *
+     * @param id       The id of the dataset to edit.
+     * @param newName  The new name of the dataset being edited.
+     */
+    @Override
+    public void editDataset(Long id, String newName) {
+
+        // filesystem path is: username/datasetId so no change is needed to the filesystem
+        Optional<Dataset> dataset = datasetRepository.findById(id);
+        if (dataset.isPresent()) {
+            dataset.get().setName(newName);
+            datasetRepository.save(dataset.get());
+        }
+    }
+
+    /**
+     * Deletes the dataset in the filesystem, then in the database.
+     *
+     * @param id The id of the dataset to delete.
+     * @param username The username of the owner of the Dataset
+     * @throws DatasetStorageException If the Dataset couldn't be deleted
+     */
+    @Override
+    public void deleteDataset(Long id, String username) throws DatasetStorageException {
+        // Delete Dataset Directories and files from the filesystem
+        try {
+            fileSystemService.deleteDatasetDirectories(id, username);
+        } catch (SecurityException e) {
+            throw new DatasetStorageException("Read access to directory was denied", e);
+        }
+        // Then delete Dataset and its files from the database
+        Optional<Dataset> dataset = datasetRepository.findById(id);
+        if (dataset.isPresent()) {
+            Set<DataFile> files = dataset.get().getFiles();
+            for (DataFile file : files) {
+                dataFileRepository.deleteById(file.getId());
```

Comment:
you can probably add a method to the datafilerepository to delete all with one called, something like `deleteByDataset`